pkg/kapis/devops/v1alpha2/devops.go

```diff
@@ -41,15 +44,81 @@ func (h *ProjectPipelineHandler) GetPipeline(req *restful.Request, resp *restful
 	resp.WriteAsJson(res)
 }
 
+func (h *ProjectPipelineHandler) getPipelinesByRequest(req *restful.Request) (api.ListResult, error) {
+	// this is a very trick way, but don't have a better solution for now
+	var (
+		err       error
+		start     int
+		limit     int
+		namespace string
+	)
+
+	// parse query from the request
+	query := req.QueryParameter("q")
+	req.Request.PostForm.Set("limit", "1000")
+	for _, val := range strings.Split(query, ";") {
+		if strings.HasPrefix(val, "pipeline:") {
+			namespace = strings.TrimLeft(val, "pipeline:")
+			namespace = strings.Split(namespace, "/")[0]
+		}
+	}
+
+	// make sure we have an appropriate value
+	if start, err = strconv.Atoi(req.QueryParameter("start")); err != nil {
+		start = 0
+	}
+	if limit, err = strconv.Atoi(req.QueryParameter("limit")); err != nil {
+		limit = 10
+	}
+
+	defer req.Request.Form.Set("limit", "10000") // assume the pipelines no more than 10k
+
+	return h.devopsOperator.ListPipelineObj(namespace, func(list []*v1alpha3.Pipeline, i int, j int) bool {
+		return strings.Compare(list[i].Name, list[j].Name) < 0
```

Comment:
Please convert list[i].Name to uppercase and compare it

Reply:
Are you suggest it should be case insensitive? But according to the naming, the name seems can only be lower-case.

Reply:
Before the versions of 3.0, the case was sensitive.

Reply:
I see.

Reply:
Fixed it https://github.com/kubesphere/kubesphere/pull/3149/commits/62c297ec53049f2e08e873ea1ee5aa3412b54d5e#diff-38024724898ee9b3e4ca7f645d4c19869b100c394ccf587ef1e1ea554f3a7a2aR76